cloudinit/config/schemas/schema-network-config-v1.json

```diff
@@ -0,0 +1,562 @@
+              "type": "string",
+              "description": "IPv4 or IPv6 gateway address."
+            },
+            "metric": {
+              "type": "integer",
+              "description": "Integer which sets the network metric value for this route."
+            }
+          }
+        },
+        "config_type_subnet": {
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "type": {
+              "type": "string",
+              "enum": ["dhcp4", "dhcp", "dhcp6", "static", "static6", "ipv6_dhcpv6-stateful", "ipv6_dhcpv6-stateless", "ipv6_slaac"]
+            },
+            "control": {
+              "type": "string",
+              "description": "Indicate how the interface will be handled during boot",
+              "enum": ["manual", "auto", "hotplug"]
+            },
+            "address": {
+              "type": "string",
+              "description": "IPv4 or IPv6 address. It may include CIDR netmask notation."
+            },
+            "netmask": {
+              "type": "string",
+              "description": "IPv4 subnet mask in dotted format or CIDR notation"
+            },
+            "gateway": {
+              "type": "string",
+              "description": "IPv4 address of the default gateway for this subnet."
+            },
+            "dns_nameservers": {
+              "type": "array",
+              "description": "List of IPv4 DNS server IPs to end up in :file:`resolve.conf`.",
+              "items": {
+                      "type": "string"
+              }
+            },
+            "dns_search": {
+              "type": "array",
+              "description": "Specify a list of search paths to be included in :file:`resolv.conf`.",
+              "items": {
+                  "type": "string"
+              }
+            },
+            "routes": {
+              "type": "array",
+              "description": "Specify a list of routes for a given interface.",
+              "items": {
+                "$ref": "#/$defs/anyOf_type_route"
+              }
+            }
+          }
+        },
+        "network_config_version1": {
+            "type": "object",
+            "additionalProperties": false,
+            "required": [
+                "config",
+                "version"
+            ],
+            "properties": {
+              "version": {
+                "type": "integer",
+                "enum": [1]
+              },
+              "config" : {
+                "type": "array",
+                "items": {
+		    "anyOf": [
+			    {"$ref": "#/$defs/anyOf_type_bond"},
+			    {"$ref": "#/$defs/anyOf_type_bridge"},
+			    {"$ref": "#/$defs/anyOf_type_physical"},
+			    {"$ref": "#/$defs/anyOf_type_nameserver"},
+			    {"$ref": "#/$defs/anyOf_type_route"},
+			    {"$ref": "#/$defs/anyOf_type_vlan"}
+                    ]
+                }
+              }
+            }
+        }
+    },
+    "required": [
```

Comment:
Hrm, right, Warn as deprecated? Or shall we just contine to permit network-config without top-level `network` key?

Reply:
Yeah, this one is a little tougher since it is coming from the clouds. Users will see a deprecation warning but can't do anything to fix it. I'm leaning towards just permitting it even though I don't like it.